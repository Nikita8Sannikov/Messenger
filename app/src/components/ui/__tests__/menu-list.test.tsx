import { render, screen } from "@testing-library/react";

import MenuListComponent from "@components/ui/menu-list";

describe("MenuListComponent", () => {
	const mockProps = {
		children: <div>Test Child</div>,
		className: "custom-class",
	};

	it("should render with children", () => {
		render(<MenuListComponent {...mockProps} />);
        
		const menuList = screen.getByRole("menu");
		expect(menuList).toBeInTheDocument();
		expect(menuList).toHaveTextContent("Test Child");
	});

	it("should apply default menu-list class", () => {
		render(<MenuListComponent {...mockProps} />);
        
		const menuList = screen.getByRole("menu");
		expect(menuList).toHaveClass("menu__list");
	});

	it("should apply custom className", () => {
		render(<MenuListComponent {...mockProps} />);
        
		const menuList = screen.getByRole("menu");
		expect(menuList).toHaveClass("custom-class");
	});

	it("should render without custom className", () => {
		render(<MenuListComponent children={mockProps.children} />);
        
		const menuList = screen.getByRole("menu");
		expect(menuList).toHaveClass("menu__list");
		expect(menuList).not.toHaveClass("custom-class");
	});

	it("should render with empty className", () => {
		render(<MenuListComponent {...mockProps} className="" />);
        
		const menuList = screen.getByRole("menu");
		expect(menuList).toHaveClass("menu__list");
		expect(menuList).not.toHaveClass("custom-class");
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<MenuListComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
