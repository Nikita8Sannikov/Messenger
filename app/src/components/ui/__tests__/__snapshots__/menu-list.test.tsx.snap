// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuListComponent matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding menu__list custom-class css-b7dj7n-MuiList-root"
    role="menu"
    tabindex="-1"
  >
    <div
      tabindex="0"
    >
      Test Child
    </div>
  </ul>
</DocumentFragment>
`;
